llvm/lib/Target/AMDGPU/AMDGPUCombinerHelper.cpp

```diff
@@ -465,13 +465,13 @@ bool AMDGPUCombinerHelper::matchCombineFmulWithSelectToLdexp(
 
     const auto SelectTrueCst =
         DestTy.isVector()
-            ? getFConstantSplat(SelectTrue, MRI, /* allowUndef */ true)
+            ? getFConstantSplat(SelectTrue, MRI, /*allowUndef=*/false)
```

Comment:
Please introduce the helper for dealing with splat or constants or drop vector support entirely

Reply:
Handled in #120935